tests/lib/hash_map_string/src/_main.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/ztest.h>
+#include <zephyr/sys/hash_map.h>
+
+#include "_main.h"
+#include "zephyr/sys/printk.h"
+#include "zephyr/sys/util.h"
+
+static uint32_t hash_string(const void *uint64_ptr_key, size_t len);
+static bool eq_string(uint64_t key_left, uint64_t key_right);
```

Comment:
I think we can probably get away with not duplicating the entire `hash_map` testsuite.

What about creating a separate testsuite inside the `tests/lib/hash_map` app with a descriptive comment or two?

E.g.

```cpp
ZTEST_SUITE(hash_map_custom_eq_func, NULL, setup, NULL, after, NULL);
```

Reply:
That makes sense, do you prefer that I conditionally compile just lines of the functions or can I just conditionally compile the whole function in the same test suite ?
Thanks in advance.

Reply:
Probably easiest to conditionally compile one ZTEST in the same testsuite. YMMV though.

Reply:
Whatever the more minimal change is. It's hard for me to say at the moment.

Reply:
Hi again @cfriedt, I tried applying the requested changes to the `_main.c` and `get.c` test files to give you an idea of what it would look like if I were to inline the `ifdef`s. It is probably a smaller change than cloning the whole function but it is also harder to read IMHO. Do you prefer it like this or cloning the whole function is ok to you ?